From 0c331d27b6213d0bfa7a3219aff31819f154ab46 Mon Sep 17 00:00:00 2001
From: kacperduras <git@kacperduras.pl>
Date: Mon, 16 Apr 2018 17:03:43 +0200
Subject: [PATCH] Remove of waiting for the server to start


diff --git a/src/main/java/org/bukkit/craftbukkit/Main.java b/src/main/java/org/bukkit/craftbukkit/Main.java
index 21628e1..1e483fa 100644
--- a/src/main/java/org/bukkit/craftbukkit/Main.java
+++ b/src/main/java/org/bukkit/craftbukkit/Main.java
@@ -203,6 +203,8 @@ public class Main {
                     System.setProperty(TerminalConsoleAppender.JLINE_OVERRIDE_PROPERTY, "false"); // Paper
                 }
 
+                // Paper start - Meleride fork, issue: GH-5
+                /*
                 if (Main.class.getPackage().getImplementationVendor() != null && System.getProperty("IReallyKnowWhatIAmDoingISwear") == null) {
                     Date buildDate = new SimpleDateFormat("yyyyMMdd-HHmm").parse(Main.class.getPackage().getImplementationVendor());
 
@@ -219,6 +221,8 @@ public class Main {
                 }
 
                 System.out.println("Loading libraries, please wait...");
+                */
+                // Paper End - Meleride fork, issue: GH-5
                 MinecraftServer.main(options);
             } catch (Throwable t) {
                 t.printStackTrace();
-- 
2.7.4

